### PR TITLE
Error info and base64

### PR DESCRIPTION
**Pull request by cursedcoder**
* Propagate status code from error
* Allow custom headers from error
* Support base64 encoded requests

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -312,4 +312,51 @@ describe('Test handling request', () => {
...
+            authorizer: undefined,
+            headers: lowercaseKeys(event.headers),
+            context: { logStreamName: '1', awsRequestId: '1' },
+          }
+        )
+      );
+
+      const error = new Error();
+      error.statusCode = 405;
+      error.headers = { 'X-API': '1.0.0' };
```
Comment (InkFaust) [inline test/index.test.js]: As I understand, error's headers are not using the same request headers, right? And it's just an optional thing.

Comment (cursedcoder) [reply test/index.test.js]: just additional headers

**Review by mark-omarov — APPROVED** [review]

File: lib/index.js [hunk 2]
```diff
@@ -5,6 +5,20 @@ const calculateExecution = endTime =>
 
 const isEmpty = v => v === null || v === undefined;
 
+const extractEventData = (event) => {
+  if (!event) {
+    return {};
+  }
+
+  if (event.body) {
+    const body = event.isBase64Encoded ? Buffer.from(event.body, 'base64').toString() : event.body;
```
Comment (dchertousov) [inline lib/index.js]: Where does flag `event.isBase64Encoded` come from?

**Review by dchertousov — COMMENTED** [review]
See my question

File: lib/index.js (hunk 2 above)
Comment (cursedcoder) [reply lib/index.js]: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-payload-encodings-workflow.html

Comment (dchertousov) [reply lib/index.js]: I don't see any mention of that flag there

Comment (cursedcoder) [reply lib/index.js]: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html

Comment (cursedcoder) [reply lib/index.js]:
```
{
    "resource": "Resource path",
    "path": "Path parameter",
    "httpMethod": "Incoming request's method name"
    "headers": {String containing incoming request headers}
    "multiValueHeaders": {List of strings containing incoming request headers}
    "queryStringParameters": {query string parameters }
    "multiValueQueryStringParameters": {List of query string parameters}
    "pathParameters":  {path parameters}
    "stageVariables": {Applicable stage variables}
    "requestContext": {Request context, including authorizer-returned key-value pairs}
    "body": "A JSON string of the request payload."
    "isBase64Encoded": "A boolean flag to indicate if the applicable request payload is Base64-encode"
}
```

Comment (cursedcoder) [reply lib/index.js]: Also see any json examples in test/ dir

Comment (dchertousov) [reply lib/index.js]: Found here: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html

**Comment by dchertousov** [comment]
Please update package minor version before merge

**Review by dchertousov — APPROVED** [review]

**Pull request merged**